src/helpers/geoChart.ts

```diff
@@ -0,0 +1,131 @@
+// (C) 2019-2020 GoodData Corporation
+import get = require("lodash/get");
+import { VisualizationObject, Execution } from "@gooddata/typings";
+import { IGeoData, IObjectMapping } from "../interfaces/GeoChart";
+import { COLOR, LOCATION, SEGMENT_BY, SIZE, TOOLTIP_TEXT } from "../constants/bucketNames";
+import {
+    getAttributeHeadersInDimension,
+    getMeasureGroupHeaderItemsInDimension,
+} from "./executionResultHelper";
+
+export function getGeoData(
+    buckets: VisualizationObject.IBucket[],
+    dimensions: Execution.IResultDimension[],
+): IGeoData {
+    const measureGroupHeader: Execution.IMeasureHeaderItem[] = getMeasureGroupHeaderItemsInDimension(
+        dimensions,
+    );
+    const attributeHeaders: Array<
+        Execution.IAttributeHeader["attributeHeader"]
+    > = getAttributeHeadersInDimension(dimensions);
+
+    const bucketItemInfos = buckets.reduce(
+        (result: IObjectMapping, bucket: VisualizationObject.IBucket): IObjectMapping => ({
+            ...result,
+            [bucket.localIdentifier]: {
+                uri:
+                    get(bucket, "items.0.visualizationAttribute.displayForm.uri") ||
```

Comment:
Please check if we have some typescript guards which would tell us if first item in bucket is attribute or measure, this would be more typescript-safe.

```
Pseudocode: 
bucketItemInfos = buckets.reduce(result, bucket) => ({
            ...result,
            [bucket.localIdentifier]: getBucketItemInfo(bucket.items[0])
})

function getFirstItemInfo(bucketItem) {
    if (isAttributeItem) {
       return { 
           uri: item.visualizationAttribute.displayForm.uri,
           localIdentifier: item.visualizationAttribute.localIdentifier
       }

       if (isMeasure...) ....
   }
}
```

Reply:
Fixed